src/compiler/moduleSpecifiers.ts

```diff
@@ -127,6 +128,15 @@ import {
     UserPreferences,
 } from "./_namespaces/ts.js";
 
+const stringToRegex = memoizeOne((pattern: string) => {
+    try {
+        return new RegExp(pattern);

```

Comment:
Do we have any other settings which are controlled by a regex like this? Trying to decide if people expect to be able to set it to a literal like `"/foo/i"` or something, but maybe that's a bit much.

That and, should this use `getRegexFromPattern` and handle case insensitive paths?

Reply:
We don’t, but C Spell does seem to allow slashes and flags. I’m not opposed to implementing that. Without the ability to provide flags, I figured case sensitive was the best option.

Reply:
Yeah, I think people can already do like `(?:i)` at the start of their patterns too?

I don't know how stable this option is or if you're just experimenting, though. That and I sort of bet people will want to write patterns that start with `/`...

Reply:
`(?:i)` isn't supported in js yet afaik so maybe better to require the whole `/.../`. Requires a little extra parsing but should be more explicit that this is a regex too

Reply:
Ah, yeah, I'm thinking of re2/Go syntax.

Other places  that take regexes are `git.branchValidationRegex` in vscode, which just passes it into `new RegExp` https://github.com/microsoft/vscode/blob/8f88d203d76dee8c92e491f8a1e1331d74792b42/extensions/git/src/commands.ts#L2679